tools/scylla-sstable.cc

```diff
@@ -2876,9 +2876,10 @@ Validate the specified sstables:
         auto& dbcfg = *pdbcfg;
         gms::feature_service feature_service(gms::feature_config_from_db_config(dbcfg));
         cache_tracker tracker;
-        dbcfg.host_id = locator::host_id::create_random_id();
```

Comment:
The scylla-sstable doesn't have token metadata indeed, but the cost of having it as large as

```
    sharded<locator::shared_token_metadata> token_metadata;
    token_metadata.start([] () noexcept { return db::schema_tables::hold_merge_lock(); }, locator::token_metadata::config{}).get();
    auto stop_token_metadata = deferred_stop(token_metadata);
```

I would try to avoid the ugliness with keeping the resolve_host_id() noncopyable_function on sstables manager and put the explicit manager->tokens dependency.

Longer term -- it's probably worth making topology _not_ be the part of token_metadata, in that case service that just need topology (e.g. -- messaging service does need it already) won't have to reference token metadata. @kbr-scylla , WDYT?

Reply:
> Longer term -- it's probably worth making topology not be the part of token_metadata, in that case service that just need topology (e.g. -- messaging service does need it already) won't have to reference token metadata. @kbr-scylla , WDYT?

Yes

Reply:
I think using a function is reasonable here.